specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -258,7 +261,135 @@ sealed partial class App : Application
+            [](
+                IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> const& sender,
+                PushNotificationCreateChannelStatus const& args)
+            {
+                if (args.status == PushNotificationChannelStatus::InProgress)
+                {
+                    // This is basically a noop since it isn't really an error state
+                    printf("The first channel request is still in progress! \n");
+                }
+                else if (args.status == PushNotificationChannelStatus::InProgressRetry)
+                {
+                    LOG_HR_MSG(
+                        args.extendedError,
+                        "The channel request is in back-off retry mode because of a retryable error! Expect delays in acquiring it. RetryCount = %d",
+                        args.retryCount);
+                }
+            });
+
+        winrt::event_token pushToken;
+
+        // Setup the completed event handler
+        channelOperation.Completed(
+            [&](
+                IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> const& sender,
+                AsyncStatus const asyncStatus)
+            {
+                auto result = sender.GetResults();
+                if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+                {
+                    auto channelUri = result.Channel().Uri();
+                    auto channelExpiry = result.Channel().ExpirationTime();
+
+                    // Register Push Event for Foreground
+                    pushToken = result.Channel().PushReceived([&](const auto&, PushNotificationReceivedEventArgs args)
+                    {
+                        auto payload = args.Payload();
+
+                        // Do stuff to process the raw payload
+
+                        // Stop the subsequent background activation from launching process again with this payload
+                        args.Handled(true);
+                    });
+
+                        // Persist the channelUri and Expiry in the App Service for subsequent Push operations
+                }
+                else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+                {
+                    LOG_HR_MSG(result.ExtendedError(), "We hit a critical non-retryable error with channel request!");
+                }
+            });
+
+        // Draw window and other foreground UI stuff here
+
+        // Unregister the Push event for Foreground before exiting
+        auto result = channelOperation.GetResults();
+        if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+        {
+            result.Channel().PushReceived(pushToken);
+        }
+    }
+
+    // To unregister Push Notifications
+    PushNotificationManager::UnregisterActivator(nullptr, pushOptions);
+    MddBootstrapShutdown(); 
```

Comment:
What is the purpose of this call?

Reply:
All this bootstrap stuff should be removed from the sample code right?

Reply:
Unless it is critical to the workings of our API in some indirect way?

Reply:
When an unpkg application exits, it removes its reference from DDLM. So that PRfwk can be serviced when all unpkg references are released.